examples/petsc/bpsdmplex.c

```diff
@@ -74,8 +74,9 @@ static int CreateRestrictionPlex(Ceed ceed, CeedInt P, CeedInt ncomp,
                                    &indices, NULL); CHKERRQ(ierr);
     for (i=0; i<numindices; i+=ncomp) {
       for (PetscInt j=0; j<ncomp; j++) {
-        if (indices[i+j] != indices[i]+j) SETERRQ1(PETSC_COMM_SELF,
-          PETSC_ERR_ARG_INCOMP, "Cell %D closure indices not interlaced", c);
+        if (indices[i+j] != indices[i] + copysign(j, indices[i]))
+          SETERRQ1(PETSC_COMM_SELF, PETSC_ERR_ARG_INCOMP,
+                   "Cell %D closure indices not interlaced", c);
       }
```

Comment:
This update seems to fix it. The check for the indices interlacing didn't account for essential BCs being encoded as `-(loc + 1)`.

Reply:
Yeah, I have a more general version of this function in `jed/ns-dmplex`. That could eventually move into a library (though it depends on both libceed and libpetsc, so either creates a dependency or needs to be in a separate library).

Reply:
Ah. I must have missed it - I got a bit turned around between all of files in that folder.

Given those options, it seems like a separate small utility library might be the way to go to prevent the extra dependency. Though if PETSc + CEED for GPU finite elements gets more traction, it might belong as part of that effort.